packages/protocol/contracts/thirdparty/LibUint512.sol

```diff
@@ -21,53 +17,10 @@ library Uint512 {
         }
     }
 
-    /// @notice Calculates the product of two uint512 and uint256
-    /// @dev Used the chinese remainder theoreme
-    /// @param a0 A uint256 representing lower bits of the first factor.
-    /// @param a1 A uint256 representing higher bits of the first factor.
-    /// @param b A uint256 representing the second factor.
-    /// @return r0 The result as an uint512. r0 contains the lower bits.
-    /// @return r1 The higher bits of the result.
-    function mul512x256(
-        uint256 a0,
-        uint256 a1,
-        uint256 b
-    ) public pure returns (uint256 r0, uint256 r1) {
-        assembly {
-            let mm := mulmod(a0, b, not(0))
-            r0 := mul(a0, b)
-            r1 := sub(sub(mm, r0), lt(mm, r0))
-            r1 := add(r1, mul(a1, b))
-        }
-    }
-
-    /// @notice Calculates the product and remainder of two uint256
-    /// @dev Used the chinese remainder theoreme
-    /// @param a A uint256 representing the first factor.
-    /// @param b A uint256 representing the second factor.
-    /// @return r0 The result as an uint512. r0 contains the lower bits.
-    /// @return r1 The higher bits of the result.
-    /// @return r2 The remainder.
-    function mulMod256x256(
-        uint256 a,
-        uint256 b,
-        uint256 c
-    ) public pure returns (uint256 r0, uint256 r1, uint256 r2) {
-        assembly {
-            let mm := mulmod(a, b, not(0))
-            r0 := mul(a, b)
-            r1 := sub(sub(mm, r0), lt(mm, r0))
-            r2 := mulmod(a, b, c)
-        }
-    }
-
-    /// @notice Calculates the difference of two uint512
-    /// @param a0 A uint256 representing the lower bits of the first addend.
-    /// @param a1 A uint256 representing the higher bits of the first addend.
-    /// @param b0 A uint256 representing the lower bits of the seccond addend.
-    /// @param b1 A uint256 representing the higher bits of the seccond addend.
-    /// @return r0 The result as an uint512. r0 contains the lower bits.
-    /// @return r1 The higher bits of the result.
+    /**
+     * Simple 512-bit addition.
+     * Taken from: https://xn--2-umb.com/17/512-bit-division/#add-subtract-two-512-bit-numbers
```

Comment:
Since there functions are all small, lets change "public" to "internal"

Reply:
ok i fixed it [d8f1822](https://github.com/taikoxyz/taiko-mono/pull/341/commits/d8f1822fa233b7953c543c1e4e6f6415cdbf0b22) and [b782f65](https://github.com/taikoxyz/taiko-mono/pull/341/commits/b782f65a5e1bcf895fd4a2854d7c495af689094a)